tests/zfs-tests/tests/functional/resilver/resilver_restart_001.ksh

```diff
@@ -98,12 +100,15 @@ VDEV_REPLACE="${VDEV_FILES[1]} $SPARE_VDEV_FILE"
 
 log_onexit cleanup
 
+# ensure that enough events will be saved
+echo 512 > $ZFS_PARAMS/zfs_zevent_len_max
```

Comment:
It's possible other occasional test failures are caused by a similar issue.  It wouldn't be unreasonable to remove the `ERPT_MAX_ERRS * MAX(max_ncpus, 4);` calculation and instead default to 512.  The existing calculation makes more sense of Solaris where it's limited to ereports, less so on Linux where it's been repurposed for generic events.  But I'm happy to leave that kind of investigation / tweak for another PR.

Reply:
I think that would be a pretty good idea. TBH, even 512 seemed a little low to me, but it is far better than 64. I do think it probably belongs in a separate PR though.